src/commands/giveaway.rs

```diff
@@ -0,0 +1,650 @@
+use chrono::{DateTime, Utc};
+use rand::seq::SliceRandom;
+use std::collections::HashMap;
+
+use serenity::{
+    builder::{CreateComponents, CreateEmbed, CreateEmbedFooter},
+    model::{
+        interactions::{
+            message_component::ButtonStyle, InteractionApplicationCommandCallbackDataFlags,
+            InteractionResponseType,
+        },
+        prelude::application_command::*,
+    },
+    prelude::*,
+};
+
+use crate::{
+    database::Database, extensions::*, models::Giveaway, Http, Interaction, Message, ReactionType,
+    User,
+};
+
+static DEFAULT_DURATION: i64 = 3600;
+static DEFAULT_WINNERS: i64 = 1;
+static DEFAULT_PRIZE: &str = "Nothing :(";
+pub static GIVEAWAY_REACTION_CHAR: char = '🎉';
```

Comment:
Could these be `const`or are they changed at some point, also these could be stored in the `.env` file.